Lib/java/std_map.i

```diff
@@ -12,60 +14,188 @@
+    java.util.Set<Entry<$typemap(jboxtype, KeyType), $typemap(jboxtype, ValueType)>> setToReturn =
+        new java.util.HashSet<Entry<$typemap(jboxtype, KeyType), $typemap(jboxtype, ValueType)>>();
+
+    Iterator itr = begin();
+    final Iterator end = end();
+    while(itr.isNot(end)) {
+      setToReturn.add(new Entry<$typemap(jboxtype, KeyType), $typemap(jboxtype, ValueType)>() {
+        private Iterator iterator;
+
+        private Entry<$typemap(jboxtype, KeyType), $typemap(jboxtype, ValueType)> init(Iterator iterator) {
+          this.iterator = iterator;
+          return this;
+        }
+
+        public $typemap(jboxtype, KeyType) getKey() {
+          return iterator.getKey();
+        }
+
+        public $typemap(jboxtype, ValueType) getValue() {
+          return iterator.getValue();
+        }
+
+        public $typemap(jboxtype, ValueType) setValue($typemap(jboxtype, ValueType) newValue) {
+          $typemap(jboxtype, ValueType) oldValue = iterator.getValue();
+          iterator.setValue(newValue);
+          return oldValue;
+        }
+      }.init(itr));
+      itr = itr.getNextUnchecked();
+    }
+
+    return setToReturn;
+  }
+%}
+
+  public:
+    map();
+    map(const map<KeyType, ValueType, Comparator >&);
+
+    struct iterator {
       %extend {
-        const T& get(const K& key) throw (std::out_of_range) {
-          std::map< K, T, C >::iterator i = self->find(key);
-          if (i != self->end())
-            return i->second;
-          else
-            throw std::out_of_range("key not found");
+        std::map<KeyType, ValueType, Comparator >::iterator getNextUnchecked() {
+          std::map<KeyType, ValueType, Comparator >::iterator copy = (*$self);
+          return ++copy;
+        }
+
+        bool isNot(const iterator other) const {
+          return (*$self != other);
         }
-        void set(const K& key, const T& x) {
-          (*self)[key] = x;
+
+        KeyType getKey() const {
+          return (*$self)->first;
         }
-        void del(const K& key) throw (std::out_of_range) {
-          std::map< K, T, C >::iterator i = self->find(key);
-          if (i != self->end())
-            self->erase(i);
-          else
-            throw std::out_of_range("key not found");
+
+        ValueType getValue() const {
+          return (*$self)->second;
         }
-        bool has_key(const K& key) {
-          std::map< K, T, C >::iterator i = self->find(key);
-          return i != self->end();
+
+        void setValue(const ValueType& newValue) {
+          (*$self)->second = newValue;
         }
       }
-  };
+    };
+
+    %rename(isEmpty) empty;
+    bool empty() const;
+    void clear();
+    iterator find(const KeyType&);
+    iterator begin();
+    iterator end();
+    %extend {
+      %fragment("SWIG_MapSize");
+
+      jint sizeImpl() const throw (std::out_of_range) {
+        return SWIG_MapSize(self->size());
+      }
 
-// Legacy macros (deprecated)
```

Comment:
Please leave all these legacy macros in. There are a number of these across different languages and I'll remove them all at the same time one day.